core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -1204,14 +1230,15 @@ public boolean equals(Object o) {
                 return false;
             }
             InFlightState that = (InFlightState) o;
-            return state == that.state && deliveryCount == that.deliveryCount;
+            return state == that.state && deliveryCount == that.deliveryCount && memberId.equals(that.memberId);
         }
 
         @Override
         public String toString() {
             return "InFlightState(" +
                 " state=" + state.toString() +
                 ", deliveryCount=" + ((deliveryCount == 0) ? "" : ("(" + deliveryCount + ")")) +
+                " memberId=" + memberId +
```

Comment:
You're missing a comma `", memberId="` is probably better.

Reply:
ahh sorry, missed it.